mtz-wizard.html

```diff
@@ -150,9 +150,11 @@
        * @return {Boolean}
        */
       validate(steps = this.steps) {
-        return (steps || []).reduce((prev, step) => {
-          return prev && (step.validate ? step.validate() : true);
-        }, true);
+        return (Promise.all((steps || []).map((step) => {
+          return step.validate ? step.validate() : Promise.resolve(true);
```

Comment:
`Promise.resolve(step.validate ? step.validate() : true)`

Reply:
Also, the steps were never updated to be promises from the looks of it.

Reply:
@admwx7 When you say the steps were never updated, what specifically do you mean? The steps are mapped to promises by the mapping functions (validate)

Reply:
https://github.com/MaritzSTL/mtz-wizard/blob/fix/validation-async/mtz-wizard-step-behavior.html#L140

Reply:
That's why i said to wrap it in a Promise.resolve

Reply:
Wouldn't it make more sense to just update all steps to resolve validate as a promise?

Reply:
If that is something we are going to require and enforce sure. As of now we don't have that as a requirement. So wrapping in promise.resolve would allow us to handle sync validate functions